haystack/document_stores/opensearch.py

```diff
@@ -515,7 +587,7 @@ def _create_document_index(self, index_name: str, headers: Optional[Dict[str, st
             if self.embedding_field:
                 index_definition["settings"]["index"] = {"knn": True}
                 # global ef_search setting affects only nmslib, for faiss it is set in the field mapping
-                if self.index_type == "hnsw":
+                if self.knn_engine == "nmslib" and self.index_type == "hnsw":
                     index_definition["settings"]["index"]["knn.algo_param.ef_search"] = 20
```

Comment:
This is probably out of scope for this PR, just a reminder that at some point we should probably allow to set custom values for ef_search (see also #3102)